compiler/rustc_mir_transform/src/deref_separator.rs

```diff
@@ -66,22 +62,8 @@ impl<'tcx> MutVisitor<'tcx> for DerefChecker<'tcx> {
                             Place::from(temp).project_deeper(&place.projection[idx..], self.tcx);
                         *place = temp_place;
                     }
-
-                    // We are destroying the previous temp since it's no longer used.
-                    if let Some(prev_temp) = prev_temp {
-                        self.patcher.add_statement(loc, StatementKind::StorageDead(prev_temp));
-                    }
```

Comment:
They don't get codegen'd but they do restrict the liveness range of variables, which makes a difference for optimizations relying on liveness and for Miri. So this is a spec change -- in the spec that we don't have, which says where and how we create storage markers.

Reply:
Ah, actually, this is in an optimization, right? Not part of codegen. So I guess spec change is the wrong term. But the following should be justified in the PR description:
- This does not affect Miri's ability to detect UB, because it is only done with optimizations enabled (right?)
- This is okay for later optimization passes that do liveness analysis, because <please fill in blank>

Reply:
I updated the pull request description a bit.

> Ah, actually, this is in an optimization, right?

No. It is part of `post_borrowck_cleanup_passes`.

> Removing storage markers is a legal program transformation. 

Those MIR passes aren't removing any storage markers.

> This does not affect Miri's ability to detect UB, because it is only done with optimizations enabled (right?)

There is no undefined behaviour by construction. Those temporaries are created for internal usage only.

Reply:
Do these temporaries ever have their address taken?

Reply:
> Do these temporaries ever have their address taken?

No.

Reply:
Ah! That makes a big difference. Please add that to the MR description, then my concerns are all resolved.